server/src/com/cloud/network/NetworkModelImpl.java

```diff
@@ -1558,6 +1558,32 @@ public boolean checkIpForService(IpAddress userIp, Service service, Long network
     }
 
     @Override
+    public boolean providerSupportsCapability(Set<Provider> providers, Service service, Capability cap) {
```

Comment:
@rhtyd  Unit test coverage added for this at  server/test/com/cloud/network/MockNetworkModelImpl.java